curvefs/test/client/chunk_cache_manager_test.cpp

```diff
@@ -0,0 +1,348 @@
+/*
+ *  Copyright (c) 2021 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: curve
+ * Created Date: Wed Mar 23 2022
+ * Author: huyao
+ */
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+#include "curvefs/src/client/s3/client_s3_adaptor.h"
+#include "curvefs/src/client/s3/client_s3_cache_manager.h"
+#include "curvefs/test/client/mock_client_s3_cache_manager.h"
+
+namespace curvefs {
+namespace client {
+using ::testing::_;
+using ::testing::DoAll;
+using ::testing::Invoke;
+using ::testing::Return;
+using ::testing::SetArgPointee;
+using ::testing::SetArgReferee;
+using ::testing::WithArg;
+
+class ChunkCacheManagerTest : public testing::Test {
+ protected:
+    ChunkCacheManagerTest() {}
+    ~ChunkCacheManagerTest() {}
+    void SetUp() override {
+        uint64_t index = 0;
+
+        S3ClientAdaptorOption option;
+        option.blockSize = 1 * 1024 * 1024;
+        option.chunkSize = 4 * 1024 * 1024;
+        option.pageSize = 64 * 1024;
+        option.intervalSec = 5000;
+        option.flushIntervalSec = 5000;
+        option.readCacheMaxByte = 104857600;
+        option.writeCacheMaxByte = 10485760000;
+        option.diskCacheOpt.diskCacheType = (DiskCacheType)0;
```

Comment:
it's better to test different parameter combinations

Reply:
agree